messari-cli/src/file_modification/substreams_yaml.rs

```diff
@@ -0,0 +1,69 @@
+use std::fs;
+use std::path::PathBuf;
+use yaml_rust::{Yaml, YamlEmitter, YamlLoader};
+
+pub(crate) struct SubstreamsYaml {
+    substreams_yaml_dir: PathBuf, // This is parent directory for the substreams.yaml file
+    yaml: Yaml,
+}
+
+impl SubstreamsYaml {
+    pub(crate) fn new(project_name: &str, substreams_filepath: &PathBuf) -> Self {
+        let substreams_yaml_dir = substreams_filepath.parent().unwrap().to_path_buf();
+
+        let yaml_contents = format!(
```

Comment:
For this kind of codegen where the content is mostly the same and we are just substituting some variables, I think a declarative approach is much more readable and maintainable. We use Mustache for subgraph templating, seems like it also supports Rust: https://github.com/nickel-org/rust-mustache